dicom_validator/validator/iod_validator.py

```diff
@@ -214,8 +241,20 @@ def _tag_id(tag_id_string):
             group = group[:2] + '00'
         return (int(group, 16) << 16) + int(element, 16)
 
+    @staticmethod
+    def _tag_id_string(tag_id):
+        tag = pydicom.tag.Tag(tag_id)
+        return str(tag).replace(' ', '')
+
     @staticmethod
     def _tag_matches(tag_value, operator, values):
+        # TODO: These fix-ups should be done in ConditionParser:
```

Comment:
Do you still want to change this?

Reply:
Ideally yes. I will try to fix it tomorrow although my day-job could interfere with my work on this validator :/

Reply:
That's ok, that's a problem most of us have :)

Reply:
I will not "fix" this in this PR. It looks like the condition parsing code has more issues than I expected. The condition parser needs to be improved/fixed in a separate PR.

With regards to the ConditionParser... the biggest issue is with tag values (such as the one above), and with conditions that are erroneously verified (i.e. the condition was parsed wrongly (incomplete, broken, etc) and thus it should have type "U", but it has some other type).

Reply:
I did respond to this yesterday, but for some reason the response is gone... Maybe I got distracted and didn't hit "Add comment".
Anyway, I do remember that the condition parser was not the finest bit of code - the whole project was more like a playground - and some refactoring/fixing  and more tests are certainly needed here. I will see that I get back to this some time in the not so far future, but meanwhile I appreciate your effort, and look forward to more PRs.

Let me know when you are ready for review for this one.

Reply:
This can be reviewed. I don't plan to work on this PR unless I find a big issue with it (unlikely because I validate with this only a handful of DCM files, and for now I just ignore the false negatives reported due to the ConditionParser issues).

(It's OK. I also work on this only when I find the time (or the need). :/
Yes, I understood from the code that it is a work-in-progress thingy. Just parsing the DICOM standard for "data" is difficult enough. So, no wonder that parsing the conditions themselves is way more difficult. But surprisingly even the current ConditionParser does a fairly decent job at it ;))

Reply:
>  and for now I just ignore the false negatives reported due to the ConditionParser issues

If you have anonymized example files for these false positives, you could write an issue for that - that could help to be sure to handle these cases later.